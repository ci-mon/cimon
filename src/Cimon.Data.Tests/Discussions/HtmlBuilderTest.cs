using AngleSharp.Dom;
using Cimon.Data.Discussions;
using Snapshooter.NUnit;

namespace Cimon.Data.Tests.Discussions;

[TestFixture]
[TestOf(typeof(HtmlBuilder))]
public class HtmlBuilderTest
{

	[Test]
	public void AddParagraph_ShouldAddExpectedContent() {
		var builder = HtmlBuilder.Create();
		builder.AddNodeWithText(TagNames.P, "test").ToString()
			.Should().MatchSnapshot();
	}

	[Test]
	public void AddSpoiler() {
		var builder = HtmlBuilder.Create();
		builder.AddSpoiler("test", htmlBuilder => {
			htmlBuilder.AddNodeWithText(TagNames.P, "yo");
		}).ToString()
			.Should().MatchSnapshot();
	}

	[Test]
	public void DoubleNestingAndAttributes() {
		using var builder = HtmlBuilder.Create();
		builder.AddNode(TagNames.P, p => {
				p["a"] = "b";
				p.AddNode(TagNames.Div, div => div.AddNodeWithText(TagNames.P, "test"));
			}).ToString()
			.Should().MatchSnapshot();
	}
}
