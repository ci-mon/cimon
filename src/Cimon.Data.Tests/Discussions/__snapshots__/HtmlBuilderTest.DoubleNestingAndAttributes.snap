<p a="b"><div><p>test</p></div></p>
